### PR TITLE
fix(mobile): remove regex lookbehind that crashes plugin load on iOS < 16.4

### LINKED ISSUE #1242: repo-invariant drift: unguarded regex lookbehind in mobile-loaded code can crash plugin load on iOS < 16.4
## What

`.claude/guidelines/coding.md` (Platform guards) states: **"Regex lookbehind is iOS 16.4+ only; add fallbacks."** A regex literal that uses lookbehind (`(?<!…)` / `(?<=…)`) is a **parse-time `SyntaxError`** on WebKit/Safari < 16.4 — not a runtime error. Because esbuild bundles the whole plugin into `main.js`, a single unsupported lookbehind literal in any eagerly-loaded module makes the entire bundle fail to parse, so the plugin **fails to load at all** on iOS < 16.4.

Two eagerly-loaded, all-platform modules currently use lookbehind with no fallback and no `Platform` guard:

1. **`src/utils/markdown-formatting.ts:108`** — `unescapeWikiLinks()` (called by `formatModelMessage()`) is on the core agent-message render path: `src/ui/agent-view/agent-view-messages.ts` calls `formatModelMessage()` at lines 130, 259, 310, 423, 705 to render every message/reasoning/plan block on every platform.
2. **`src/services/ollama-models-service.ts:12–15`** — module-level `COMPLETION_NAME_HINT_PATTERNS` array. `ModelManager` (instantiated in `src/main.ts:154`, a core service loaded on all platforms) imports this module eagerly via `src/services/model-manager.ts:6`, so these top-level regex literals are parsed at plugin load on mobile too.

Both sites share one root cause: lookbehind in code that parses on mobile, against the documented iOS-16.4 rule.

## Evidence

- `src/utils/markdown-formatting.ts:108` — `segment.replace(/(?<!`)`(\[\[[^\]]+\]\])`(?!`)/g, '$1')` (negative lookbehind to avoid matching multi-backtick spans).
- `src/services/ollama-models-service.ts:12` — `/(?<!\d)0\.5b(?!\d)/i`
- `src/services/ollama-models-service.ts:13` — `/(?<!\d)1\.5b(?!\d)/i`
- `src/services/ollama-models-service.ts:14` — `/(?<!\d)1b(?!\d)/i`
- `src/services/ollama-models-service.ts:15` — `/(?<!\d)3b(?!\d)/i`
- Import chain (all eager, top-level): `src/main.ts:13` → `src/services/model-manager.ts:6` → `src/services/ollama-models-service.ts` (module-level regex literals).
- Rule being violated: `.claude/guidelines/coding.md` → **Platform guards** → "Regex lookbehind is iOS 16.4+ only; add fallbacks."

## Proposed unit of work

Eliminate the lookbehind at both sites with behavior-preserving, lookbehind-free equivalents, and add a regression test for each:

- **`markdown-formatting.ts`** — rewrite the single-backtick-wrap strip to capture the preceding character instead of asserting it, e.g. `/(^|[^`])`(\[\[[^\]]+\]\])`(?!`)/g` with replacement `'$1$2'` (the trailing `(?!`)` lookahead is universally supported and can stay). Add tests to `test/utils/markdown-formatting.test.ts` (if absent, create it) covering: backtick-wrapped wikilink unwrapped; double-backtick span left intact; adjacency edge case (two wrapped links in a row).
- **`ollama-models-service.ts`** — replace the four digit-boundary lookbehind patterns with a non-lookbehind form (e.g. `/(^|\D)0\.5b(?!\d)/i` matched against the normalized name, or a small helper that checks the char before the match) so `1b` still doesn't bleed into `11b`. Cover with a unit test asserting `llava:13b` is not biased toward completions while `qwen:1b` is.
- **Prevent recurrence (systemic):** the rule already exists in `coding.md` and is still being violated, so the gap is enforcement, not documentation. Add a mechanical guard — an ESLint rule or a CI grep (e.g. fail on `(?<` in `src/**/*.ts` outside a documented desktop-only, `Platform`-guarded allowlist) — so the next lookbehind literal is caught at review/CI time rather than by a mobile user's crash report.

## Out of scope

- Any change to the wikilink-unescaping or Ollama model-biasing *behavior* — this is a pure lookbehind-elimination + guard task.
- The `(?!…)` lookaheads, which are supported everywhere and need no change.
- Splitting or otherwise refactoring the affected files (tracked separately for oversized files under #801).

---

*Filed by the `audit-architecture` skill. If this isn't worth doing, close as **Not planned** / `wontfix` — the skill checks closed-not-planned and won't refile.*

### PR DESCRIPTION
**Pull request by allenhutchison**
## Summary

`.claude/guidelines/coding.md` (Platform guards) requires fallbacks for regex lookbehind because it is only supported on **iOS 16.4+**, and `manifest.json` sets `isDesktopOnly: false` — the plugin loads on mobile. Three negative-lookbehind regex **literals** were bundled into `main.js`. On iOS < 16.4 an unsupported lookbehind in a literal is a parse-time `SyntaxError` when the module is evaluated, which aborts the whole bundle — so this crashes plugin **load**, not just the affected feature.

This PR rewrites the three sites to a lookbehind-free consuming-group form. Lookaheads are left intact (Safari has supported them for years; only lookbehind is the 16.4 gate), and a source-scan guard test prevents any lookbehind literal from silently returning.

Fixes #1242

Produced by the auto-dev pipeline from the approved plan on #1242 (`auto:ready`).

## Changes

- `src/utils/markdown-formatting.ts` — the backtick-wrapped-wikilink strip pattern now uses a leading `(^|[^`])` group with replacement `$1$2` instead of `(?<!`)`, preserving the "not preceded by a backtick" semantics. The `(?!`)` lookahead is kept.
- `src/services/ollama-models-service.ts` — the four small-model size-hint patterns (`0.5b`/`1.5b`/`1b`/`3b`) now use a `(?:^|\D)` prefix instead of `(?<!\d)`. Safe because these patterns feed `.test()` (`COMPLETION_NAME_HINT_PATTERNS.some((re) => re.test(lower))`), so consuming a preceding non-digit does not change the boolean result. The `(?!\d)` lookahead is kept, so `1b` still doesn't bleed into `11b` and `llava:13b` isn't biased toward completions.
- `test/utils/no-regex-lookbehind.test.ts` (new) — scans `src/**/*.ts` and fails on any `(?<=` / `(?<!` literal, enforcing the guideline's platform guard automatically (mirrors the existing `test/services/mobile-node-builtins.test.ts` source-scan guard pattern).
- `test/utils/markdown-formatting.test.ts` — added edge cases confirming the consuming group preserves the preceding character and still handles start-of-string.
- `test/services/ollama-models-service.test.ts` — added a boundary block asserting genuine small models (`0.5b`/`1.5b`/`1b`/`3b`) get the `completions` role while `13b`/`11b`/`23b` do not.

## Screenshots / Screencast

N/A — internal regex refactor, no UI surface.

## Verification

- **Pre-flight (all green):** `npm run format-check`, `npm run typecheck:test`, `npm run lint` (0 errors), `npm test` (3523 passed), `npm run build`, `npm run lint:cycles` (no circular deps).
- **Behavioral (semantics):** both changed code paths are exercised end-to-end by unit tests — the markdown formatter (backtick-wrapped wikilinks still stripped, multi-backtick spans untouched) and the ollama classifier (`getModels() → defaultForRoles`, small-model boundaries preserved, `13b`/`11b` excluded). The new guard test confirms zero lookbehind literals remain under `src/`.
- **Platform (iOS < 16.4 load):** the crash itself cannot be reproduced in the headless Node sandbox (no old-iOS device) — **behavioral verification not run in sandbox; needs a manual check on an iOS < 16.4 device** to confirm plugin load. The fix is structurally guaranteed by removing all lookbehind literals and enforced going forward by the new source-scan guard.

### Required

- [x] I have read and agree to the [Contributing Guidelines](../CONTRIBUTING.md)
- [x] I have read and agree to the [AI Policy](../AI_POLICY.md)
- [x] This PR is linked to an approved issue where the approach was discussed with a maintainer (#1242, plan approved via `auto:ready`)
- [x] All CI checks pass (`npm test`, `npm run build`, `npm run format-check`; also `npm run typecheck:test`, `npm run lint`, `npm run lint:cycles`)
- [ ] I have tested this change on Desktop — desktop is unaffected (lookbehind works on desktop Electron); the fix targets iOS < 16.4.
- [x] I have verified this change does not break Mobile (or includes appropriate platform guards) — the change *is* the platform guard: it removes the unsupported syntax and adds a guard test. Semantics preserved via unit tests; live iOS < 16.4 load confirmation still needs a device (see Verification).
- [x] Documentation has been updated (if applicable) — N/A; the guard is already stated in `.claude/guidelines/coding.md` and the new test enforces it.
- [x] I understand that I must address all review comments from CodeRabbit and maintainers, or this PR may be closed

### AI-Generated Code

- [x] This PR includes AI-generated or AI-assisted code
- [x] AI tool(s) used: Claude (auto-dev pipeline)
- [x] I have reviewed and understand all AI-generated code in this PR

This PR was produced by the auto-dev pipeline from the approved plan on #1242.

---
_Generated by [Claude Code](https://claude.ai/code/session_01FwL6EPwB8Hw6a131XLXRFn)_

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved model-name classification to prevent larger models from being incorrectly identified as small-model completions.
  * Fixed WikiLink formatting in edge cases, including links at the start of text and links without preceding spaces.
  * Improved compatibility with older iOS versions by removing unsupported regular-expression features.

* **Tests**
  * Added regression coverage for model-name matching, WikiLink formatting, and unsupported regular-expression patterns.

### CODE REVIEW

File: src/services/ollama-models-service.ts [hunk 1]
```diff
@@ -7,12 +7,18 @@ import { GeminiModel } from '../models';
...
  */
 const COMPLETION_NAME_HINT_PATTERNS = [
-	/(?<!\d)0\.5b(?!\d)/i,
-	/(?<!\d)1\.5b(?!\d)/i,
-	/(?<!\d)1b(?!\d)/i,
-	/(?<!\d)3b(?!\d)/i,
+	/(?:^|\D)0\.5b(?!\d)/i,
+	/(?:^|\D)1\.5b(?!\d)/i,
+	/(?:^|\D)1b(?!\d)/i,
+	/(?:^|\D)3b(?!\d)/i,
```
[the remark this answers is not in the document]
Comment (allenhutchison) [reply src/services/ollama-models-service.ts:21]: Declining this one — the finding is based on a misread of the convention. The repo's Prettier style is **tabs**, not spaces: `.prettierrc` sets `tabWidth: 2` while `.editorconfig` sets `indent_style = tab`, and Prettier honors editorconfig's `useTabs`, so the enforced style is tab characters displayed at 2 columns. All five changed blocks use tabs to match, and `npm run format-check` (`prettier --check`) passes on every one of them — it also ran via the precommit `prettier --write` on this exact commit. Converting them to literal two-space indentation would fail `format-check` and break the `lint` CI job. The AGENTS.md "2-space indent" line describes the rendered width, not literal spaces.

---
_Generated by [Claude Code](https://claude.ai/code)_
